File: kotlin-lambda/src/main/kotlin/jp/ne/naokiur/handlers/dynamodb/DynamoDBManager.kt
package jp.ne.naokiur.handlers.dynamodb

import com.amazonaws.regions.Regions
import com.amazonaws.services.dynamodbv2.AmazonDynamoDB
import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder
import com.amazonaws.services.dynamodbv2.model.AttributeValue
import com.amazonaws.services.dynamodbv2.model.PutItemRequest
import com.amazonaws.services.dynamodbv2.model.PutItemResult
import java.util.*

class DynamoDBManager {
    private val dynamodb: AmazonDynamoDB = AmazonDynamoDBClientBuilder
            .standard()
            .withRegion(Regions.AP_NORTHEAST_1)
            .build()

    private val tableName = "login-history"

    fun putItem(userName: String, loginTime: Calendar): PutItemResult {

        // Convert milli time to time.
        val time: Long = loginTime.time.time / 1000

        val item = mapOf(
                "user_name" to AttributeValue(userName),
                "login_time" to AttributeValue().withN(time.toString())
        )

        val putItemRequest = PutItemRequest(tableName, item)
        return dynamodb.putItem(putItemRequest)
    }
}

File: python-lambda/handlers/dynamodb/dynamodb_manager.py
from datetime import datetime
import boto3


class DynamodbManager:
	def __init__(self):
		self.dynamodb = boto3.resource(
				'dynamodb',
				region_name='ap-northeast-1'
		)
		self.table_name = 'login-history'

	def put_item(self, user_name: str, login_time: datetime):
		table = self.dynamodb.Table(self.table_name)
		res = table.put_item(
				Item={
					'user_name': user_name,
					'login_time': int(login_time.timestamp())
				}
		)

		return res
